cmd/install.go

```diff
@@ -108,19 +94,58 @@ func getUserList(sysUsers v1beta1.SystemUser) []string {
 }
 
 func getCmdArgs(role string) []string {
+	params := make(map[string]string)
+
+	params["data-dir"] = dataDir
+
 	var args []string
 
 	if role == "server" {
 		args = append(args, "server")
+
+		//server params
+		if serverWorkerProfile != "default" {
+			params["profile"] = serverWorkerProfile
+		}
+
+		if enableWorker {
+			params["enable-worker"] = ""
+		}
+
+		params["token-file"] = serverToken
+
 		if cfgFile != "" {
 			args = append(args, "--config", cfgFile)
 		}
 	} else {
-		args = append(args, "worker", "--token-file", "REPLACEME")
+		args = append(args, "worker")
+
+		//worker params
+		if workerProfile != "default" {
+			params["profile"] = workerProfile
+		}
+
+		if tokenFile != "" {
+			params["token-file"] = tokenFile
+		} else {
+			params["token-file"] = "REPLACEME"
+		}
+
+		params["cri-socket"] = criSocket
 	}
 
 	if debug {
 		args = append(args, "--debug")
 	}
+
+	for k, v := range params {
+		if v != "" {
+			args = append(args, (fmt.Sprintf("--%s %s", k, v)))
+		}
+		if k == "enable-worker" || k == "enable-cloud-provider" {
```

Comment:
Couldn't this just be:

```suggestion
		} else {
```

Reply:
yes it could